### PR TITLE
Avoid killing Flutter tool process (#37471)

### LINKED ISSUE #37471: Flutter tool process is killed when using use-application-binary option
When using the `use-application-binary` option with `flutter run` there is some code that kills the flutter tool running process. 

For example:

`flutter run --use-application-binary my_executable_path`

When this code https://github.com/flutter/flutter/blob/3fedb8cb5ada929a697f29fb1e5783f9cfa76a15/packages/flutter_tools/lib/src/desktop.dart#L13 runs from https://github.com/flutter/flutter/blob/d6bd1c0584d761e622c9f9331581bcae3655f0c4/packages/flutter_tools/lib/src/linux/linux_device.dart#L90, any process that includes the string `my_executable_path` is killed, even the one that is running the flutter tool, since the string is passed as an argument.

My suggestion would be to exclude the current running process when looking for processes to kill using https://api.dartlang.org/stable/2.4.0/dart-io/pid.html

### PR DESCRIPTION
**Pull request by davidmartos96**
## Related Issues

Fixes #37471

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Hi @davidmartos96 , this will need unit tests before we can review it. For examples, see the existing tests in desktop_test.dart and windows_test.dart

**Comment by davidmartos96** [comment]
@jonahwilliams I included some tests. Reproducing the related issue in Windows is very rare in a real case scenario, but I added a test case for it anyway.

**Comment by jonahwilliams** [comment]
@davidmartos96  its not about reproducing the exact scenarios, but making sure that the intention of the code is documented in a reproducible way. Imagine that someone were refactoring this in a year. They might not know what the continue is for, or otherwise forget it. Then we end up regressing without a test case that documents the purpose.

**Comment by davidmartos96** [comment]
@jonahwilliams Yes, I agree with you. The tests I added would fail if the new `continue`s are removed. Do you think the PR needs more changes?
I should mention that I couldn't run the macos_device_test because I don't have a Mac. The test case is very similar to the Linux one though.

**Comment by jonahwilliams** [comment]
Off the top of my head, the tests look sufficient. I can verify the macOS case locally. In the meantime, please fix the errors flagged by the analysis task. That will let the code coverage shard run.

File: packages/flutter_tools/lib/src/desktop.dart [hunk 1]
```diff
@@ -29,9 +29,15 @@ Future<bool> killProcess(String executable) async {
       if (values.length < 2) {
         continue;
       }
-      final String pid = values[1];
+      final String processPid = values[1];
+      final String currentRunningProcessPid = pid.toString();
+      // Don't kill the flutter tool process
+      if (processPid == currentRunningProcessPid){
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/desktop.dart]:
```suggestion
      if (processPid == currentRunningProcessPid) {
```

File: packages/flutter_tools/lib/src/windows/windows_device.dart [hunk 2]
```diff
@@ -174,8 +174,15 @@ List<String> runningProcess(String processName) {
       continue;
     }
     final List<String> parts = process.split(_whitespace);
+
+    final String processPid = parts[0];
+    final String currentRunningProcessPid = pid.toString();
+    // Don't kill the flutter tool process
+    if (processPid == currentRunningProcessPid){
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/windows/windows_device.dart]:
```suggestion
    if (processPid == currentRunningProcessPid) {
```

**Review by jonahwilliams — APPROVED** [review]
LGTM, with two small nits. You'll also need to pull to handle a file diff.

In the future when you update a PR, please @ me or another review in the comments so I know to take a look, otherwise I get no notification from Github and it drops off my radar.

**Comment by christopherfujino** [comment]
@davidmartos96 everything looks good to go...except there's a merge conflict. Sorry, but can  you resolve? Then I'd be happy to merge it.

**Comment by davidmartos96** [comment]
@jonahwilliams Merge conflict resolved with code styling fixes

**Comment by christopherfujino** [comment]
Restarting cirrus tests that timed out....

**Comment by davidmartos96** [comment]
@christopherfujino Build seems good now

**Pull request merged**